src/ocean/task/TaskPool.d

```diff
@@ -175,12 +180,16 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
             item.terminationHook({
                 --count;
                 if ((count == 0) && task.suspended())
+                {
                     task.resume();
+                }
             });
         }
 
         if (count > 0)
+        {
```

Comment:
You really don't to check if it is suspended at that point, it can only be otherwise in case of programming error so better plain `resume` and let assert fire.